internal/commands/app/new_app_inputs.go

```diff
@@ -0,0 +1,71 @@
+package app
+
+import (
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/atlas"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/terminal"
+)
+
+const (
+	flagProject      = "project"
+	flagProjectUsage = "the MongoDB cloud project id"
+
+	flagFrom      = "from"
+	flagFromShort = "a"
+	flagFromUsage = "choose an application or template to initialize the Realm app with"
+
+	flagName      = "name"
+	flagNameShort = "n"
+	flagNameUsage = "set the name of the Realm app to be initialized"
+
+	flagDeploymentModel        = "deployment-model"
+	flagDeploymentModelShort   = "d"
+	flagDeploymentModelUsage   = `select the Realm app's deployment model, available options: ["global", "local"]`
+	flagDeploymentModelDefault = realm.DeploymentModelGlobal
+
+	flagLocation        = "location"
+	flagLocationShort   = "l"
+	flagLocationUsage   = `select the Realm app's location, available options: ["US-VA", "local"]`
+	flagLocationDefault = realm.LocationVirginia
+)
+
+type from struct {
+	GroupID string
+	AppID   string
+}
+
+func (f from) IsZero() bool {
+	return f.GroupID == "" && f.AppID == ""
+}
+
+type newAppInputs struct {
+	Project         string
+	From            string
+	Name            string
+	DeploymentModel realm.DeploymentModel
+	Location        realm.Location
+}
+
+func (i *newAppInputs) resolveFrom(ui terminal.UI, rc realm.Client, ac atlas.Client) (from, error) {
+	var f from
+
+	if i.From != "" {
+		if i.Project == "" {
+			groupID, groupErr := cli.ResolveGroupID(ui, ac)
+			if groupErr != nil {
+				return from{}, groupErr
+			}
+			i.Project = groupID
+		}
+
+		app, err := cli.ResolveApp(ui, rc, realm.AppFilter{GroupID: i.Project, App: i.From})
```

Comment:
I think we're conflating `Project` and `From` here a little bit

`Project` is meant to be the `GroupID` we use to specify/make changes to a Realm app, the target group if you will.  it also serves as a first "filter" for when we fetch available apps to the user (as mongo employees, we have a ton of available apps to us).

`From` is simply an app that the user has access to and wants to read data from (in this case to create a similarly structured app).  we can use the `i.Project` flag as an optional filter, but we don't want to "resolve` it until we need to.

for instance, `app init` never _needs_ to resolve the project/group id to write to.
however, `app create`, `push` and `pull` will.

knowing that, i think the `if i.Project == ""` chunk of logic above can probably be moved to its own function on `(i *newAppInputs)`, so that those commands can call it directly when need be in their own handlers/resolves